### PR TITLE
[SYCLomatic][CMake] Add support `CUDA_HOST_COMPILER`, `CUDA_HOST_FLAGS`

### PR DESCRIPTION
**Pull request by the-slow-one**
We are creating an equivalent of the two listed variables in dpct.cmake. And these variables will only replace the occurance of CUDA_HOST_COMPILER and CUDA_HOST_FLAGS. The variables will be used in `cuda_execute_process`, another API, which we will support in future.

### CODE REVIEW

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by TejaX-Alaghari — APPROVED** [review]
LGTM

File: clang/tools/dpct/cmake/dpct.cmake [hunk 1]
```diff
@@ -119,4 +119,6 @@ set(CMAKE_SYCL_FLAGS "")
 set(COMPATIBILITY_VERSION 99.9)
 set(COMPATIBILITY_VERSION_MAJOR 99)
 set(COMPATIBILITY_VERSION_MINOR 9)
-set(COMPATIBILITY_VERSION_STRING "${COMPATIBILITY_VERSION}")
\ No newline at end of file
+set(COMPATIBILITY_VERSION_STRING "${COMPATIBILITY_VERSION}")
+set(SYCL_HOST_COMPILER "icpx")
```
Comment (tomflinda) [inline clang/tools/dpct/cmake/dpct.cmake:122]: Unify Line115 and Line123:
set(SYCL_HOST_COMPILER "icpx")

Comment (the-slow-one) [reply clang/tools/dpct/cmake/dpct.cmake:122]: Done

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**